### ISSUE TITLE
Can't type space on iOS

### ISSUE THREAD

**Issue by CorcovadoMing** [paradoxxxzero/butterfly#108]
I can't type space on my iPad (iOS 9.3.2) with external keyboard

**Comment by shimaode** [comment]
Me too, and only unsecured way can be used to see the terminal, I can't get SSL way working even if I installed ca and p12 key on the phone, I can only see a black screen. Everything is fine on PC.

**Comment by shimaode** [comment]
I fixed this issue by modify line 1634 of main.js
from:
`if (!key) {`
to:
`if (!key && !(ev.keyCode==32 && navigator.platform=="iPad")) {`

But I don't know how to modify the corresponding term.coffee, so I didn't commit a pull request.
By the way, SSL is still not working on mobile devices.

**Comment by CorcovadoMing** [comment]
@shimaode I tried to modify main.js in official docker container located in `butterfly/static/main.js` and copy the `butterfly/static/main.js` to `butterfly/static/main.min.js`. It did change the behavior when I type space on iPad. However, it didn't correctly show up the space but `undefined` instead

**Comment by CorcovadoMing** [comment]
@shimaode I know nothing about coffeescript, but I assume it is related to https://github.com/paradoxxxzero/butterfly/blob/master/coffees/term.coffee#L1526

**Comment by shimaode** [comment]
@CorcovadoMing   Sorry I made a mistake, my bad. I modified main.min.js directly and wrote the wrong version of corresponding main.js
It should be
      if (ev.keyCode==32 && navigator.platform=="iPad")
        key=" "
      else if (!key) {
        return true;
      }
      this.showCursor();
      this.send(key);
      return cancel(ev);

**Comment by shimaode** [comment]
@CorcovadoMing, you are right, it is related to term.coffee. Have you got your ssl version of butterfly working on your iPad?

**Comment by CorcovadoMing** [comment]
@shimaode Sorry, I didn't use SSL. Instead, I used login username and password as in docker section in README
